### PR TITLE
Feature/upgrade contracts versions for load and notary

### PR DESCRIPTION
**Pull request by jianwel**
ready for review.

### CODE REVIEW

**Comment by ajhodges** [comment]
@jianwel looks like there are some conflicts in the postman collection, can you please take a look/rebase?

File: src/__tests__/contracts.ts [hunk 1]
```diff
@@ -66,11 +67,14 @@ export const ContractEntityTests = async function() {
             const [signed_tx, txHash] = await owner.sign_tx(txParams);
 
             const receipt: any = await network.send_tx(signed_tx);
+            console.log(receipt.transactionHash.length)
 
             expect(receipt.transactionHash.length).toEqual(66);
 
             const new_owner_balance = await local.ShipToken.call_static('balanceOf', [owner.address]);
             const new_other_balance = await local.ShipToken.call_static('balanceOf', [other.address]);
+            console.log(new_owner_balance)
```
Comment (mlclay) [inline src/__tests__/contracts.ts]: Please remove debug log statements

File: src/__tests__/contracts.ts [hunk 2]
```diff
@@ -22,11 +22,12 @@ import { Contract, Version, Project, Network } from '../entity/Contract';
 import { EncryptorContainer } from '../entity/encryption/EncryptorContainer';
 import { EthereumService } from "../eth/EthereumService";
 
+import { latest as LATEST_NOTARY } from '../../rpc/VaultNotary/Latest';
```
Comment (mlclay) [inline src/__tests__/contracts.ts]: For consistency, let's define this one near line 30 and hard code the version `1.0.0` like the other LATEST_* constants.  Since we are explicitly testing specific versions of the contracts, we don't actually want this to import the latest version like we do elsewhere.

File: docs/LoadContract_1.2.0.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+}
+```  
+We can see the only difference is the added `carrierWallet`, compared with the parameter
+ list of this endpoint in the version 1.1.0. 
+  
+2\. Removed all the methods/endpoints related to the uri and hash. Those functions are now separated
+into a new Notary contract. The information about the rpc calls to this new contract 
+can be found at  [VaultNotary 1.0.0](./VaultNotaryContract_1.0.0.md).
+ 
+ This is a list the removed endpoints,
```
Comment (ajhodges) [inline docs/LoadContract_1.2.0.md]:
```suggestion
 This is a list of the removed endpoints,
```

File: docs/VaultNotaryContract_1.0.0.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+		"vaultId": "{{vault_id}}",
+		"senderWallet": "{{wallet_id}}", 
+		"toRevokeWallet": "bf6204a5-5e1a-458c-a180-c54cc35d05a2"
+	},
+	"jsonrpc": "2.0",
+	"id": 0
+}
+```
+
+This will revoke the permission to update the URI of the vault with `vaultId` to
```
Comment (ajhodges) [inline docs/VaultNotaryContract_1.0.0.md]:
```suggestion
This will revoke the permission to update the URI of the vault with `vaultId` from 
```

File: docs/VaultNotaryContract_1.0.0.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
```
Comment (ajhodges) [inline docs/VaultNotaryContract_1.0.0.md]:
```suggestion
This will revoke the permission to update the URI of the vault with `vaultId` from 
```

File: rpc/Load/1.2.0/RPCLoad.ts [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+        require: ['shipmentUuid', 'senderWallet', 'depositAmount'],
+        validate: {
+            uuid: ['shipmentUuid', 'senderWallet'],
+        },
+    })
+    public static async FundEscrowTx(args) {
+        const senderWallet = await Wallet.getById(args.senderWallet);
+
+        const LOAD_CONTRACT: LoadContract = <LoadContract>loadedContracts.get(PROJECT, VERSION);
+        const TOKEN_CONTRACT: ShipTokenContract = <ShipTokenContract>loadedContracts.get('ShipToken');
```
Comment (ajhodges) [inline rpc/Load/1.2.0/RPCLoad.ts]: My intuition is that the `'ShipToken'` string should be pulled out into a constant but I'm not sure where it should live. @mlclay?

File: docs/VaultNotaryContract_1.0.0.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+{
+    "jsonrpc": "2.0",
+    "id": 0,
+    "result": {
+        "success": true,
+        "details": {
+            "0": "s3://my-bucket/2ed96ba9-26d4-4f26-b3da-c45562268480/meta.json",
+            "1": "0x0f021e716c58d1e53222bf1cf9dfe8740470ef8d4a125499b26de58049688563",
+            "vaultUri": "s3://my-bucket/2ed96ba9-26d4-4f26-b3da-c45562268480/meta.json",
+            "vaultHash": "0x0f021e716c58d1e53222bf1cf9dfe8740470ef8d4a125499b26de58049688563"
```
Comment (ajhodges) [inline docs/VaultNotaryContract_1.0.0.md]: Update the example with the new response

Comment (jianwel) [reply docs/VaultNotaryContract_1.0.0.md]: fixed

File: docs/LoadContract_1.2.0.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+   Now the new `create_shipment_tx` should be called as follows,
+  
+```JSON
+{
+	"method": "load.1.2.0.create_shipment_tx",
+	"params": {
+		"shipmentUuid": "77777777-25fe-465e-8458-0e9f8ffa2cdd",
+		"senderWallet": "8ca1e0df-26ff-4331-86f7-61a1cdcb06c9", 
+		"fundingType": 1, 
+		"contractedAmount": 10000000000000,
```
Comment (mlclay) [inline docs/LoadContract_1.2.0.md]: This needs to be a String value now, correct?

File: rpc/Engine.postman_collection.json [hunk 9]
```diff
@@ -1986,183 +1993,728 @@
...
+										"method": "POST",
+										"header": [
+											{
+												"key": "Content-Type",
+												"value": "application/json"
+											}
+										],
+										"body": {
+											"mode": "raw",
+											"raw": "{\n\t\"method\": \"load.1.2.0.create_shipment_tx\",\n\t\"params\": {\n\t\t\"shipmentUuid\": \"{{shipment_uuid}}\",\n\t\t\"senderWallet\": \"{{wallet_id}}\", \n\t\t\"fundingType\": 1, \n\t\t\"contractedAmount\": 10000000000000, \n\t\t\"carrierWallet\": \"{{wallet_id}}\"\n\t},\n\t\"jsonrpc\": \"2.0\",\n\t\"id\": 0\n}"
```
Comment (mlclay) [inline rpc/Engine.postman_collection.json]: Since we updated the number parameters to strings, make sure they are updated in the Postman Collection as well

File: rpc/VaultNotary/1.0.0/RPCVaultNotary.ts [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+const loadedContracts = LoadedContracts.Instance;
+const PROJECT = 'NOTARY';
+const VERSION = '1.0.0';
+
+@RPCNamespace({ name: 'VaultNotary.1.0.0' })
+export class RPCVaultNotary {
+    @RPCMethod({
+        require: ['vaultId', 'senderWallet', 'vaultUri', 'vaultHash'],
+        validate: {
+            uuid: ['vaultId', 'senderWallet'],
```
Comment (mlclay) [inline rpc/VaultNotary/1.0.0/RPCVaultNotary.ts]: Do we want to validate that `vaultUri` and `vaultHash` are strings for the requests to the Notary contract?

File: rpc/__tests__/notary.ts [hunk 11]
```diff
@@ -0,0 +1,426 @@
...
+                expect(rpcReturn.transaction.gasPrice).toBeDefined();
+                expect(rpcReturn.transaction.gasLimit).toBeDefined();
+                expect(rpcReturn.transaction.value).toBeDefined();
+                expect(rpcReturn.transaction.data).toBeDefined();
+                expect(rpcReturn.transaction.to).toBeDefined();
+                expect(rpcReturn.transaction.chainId).toBeDefined();
+                expect(rpcReturn.contractVersion).toMatch(LATEST_NOTARY);
+ 
+        }));
+
```
Comment (mlclay) [inline rpc/__tests__/notary.ts]: If you validate the vaultUri and vaultHash RPC parameters as Strings, put some tests in here where you provide non-string values.  There should be example tests that utilize the `expectInvalidStringParams` test assertion helper.

File: rpc/contracts.ts [hunk 12]
```diff
@@ -156,17 +166,23 @@ export async function loadContractFixtures() {
...
     await TOKEN_CONTRACT.Ready;
     await LOAD_CONTRACT.Ready;
+    await NOTARY_CONTRACT.Ready;
 
     loadedContracts.register('LOAD', LOAD_CONTRACT, true);
     loadedContracts.register('ShipToken', TOKEN_CONTRACT, true);
+    loadedContracts.register('NOTARY', NOTARY_CONTRACT, true);
 
-    await registerPreviousLoadContracts(contractMetaData.LOAD.deployed[network], LOAD_CONTRACT);
+    await registerPreviousLoadContracts(contractMetaData.LOAD.deployed['mainnet'], LOAD_CONTRACT);
```
Comment (mlclay) [inline rpc/contracts.ts]: Undo this change we made when debugging

File: rpc/__tests__/notary.ts (hunk 11 above)
Comment (jianwel) [reply rpc/__tests__/notary.ts]: This test assertion helper is very handy. Thanks!
